ArduPlane/quadplane.cpp

```diff
@@ -4577,4 +4610,25 @@ bool QuadPlane::should_disable_TECS() const
     return false;
 }
 
+// calculate limit forward tilt demand and calculate throttle required to compensate
+// calculate forward tilt demand limit and calculate throttle required to compensate
+void QuadPlane::calc_fwd_tilt_throttle()
+{
+    float fwd_tilt_rad = radians(constrain_float(-0.01f * (float)plane.nav_pitch_cd, 0.0f, 45.0f));
+    q_fwd_throttle = MIN(plane.quadplane.q_fwd_thr_gain * tanf(fwd_tilt_rad), 1.0f);
+
+    // To prevent forward motor prop strike, reduce throttle to zero when close to ground
+    // When we are doing horizontal positioning in a VTOL land
+    // we always allow the fwd motor to run. Otherwise a bad
+    // lidar could cause the aircraft not to be able to
+    // approach the landing point when landing below the takeoff point
+    if (!in_vtol_land_approach()) {
+        float alt_cutoff = MAX(0,vel_forward_alt_cutoff);
+        float height_above_ground = plane.relative_ground_altitude(plane.g.rangefinder_landing);
+        float fwd_thr_scaler = linear_interpolate(0.0f, 1.0f, height_above_ground, alt_cutoff, alt_cutoff+2);
+        q_fwd_throttle *= fwd_thr_scaler;
+    }
+
+    q_fwd_nav_pitch_lim_cd = MIN(-plane.quadplane.q_fwd_tilt_lim, 0);
```

Comment:
Need to relax the limit as we reduce the throttle for the low altitude case.

Reply:
Done